src/view/writer.js

```diff
@@ -12,704 +12,993 @@ import ContainerElement from './containerelement';
-}
 
-/**
- * Moves nodes from provided range to target position.
- *
- * Throws {@link module:utils/ckeditorerror~CKEditorError CKEditorError} `view-writer-invalid-range-container` when
- * {@link module:engine/view/range~Range#start start} and {@link module:engine/view/range~Range#end end} positions are not placed inside
- * same parent container.
- *
- * @function module:engine/view/writer~writer.move
- * @param {module:engine/view/range~Range} sourceRange Range containing nodes to move.
- * @param {module:engine/view/position~Position} targetPosition Position to insert.
- * @returns {module:engine/view/range~Range} Range in target container. Inserted nodes are placed between
- * {@link module:engine/view/range~Range#start start} and {@link module:engine/view/range~Range#end end} positions.
- */
-export function move( sourceRange, targetPosition ) {
-	let nodes;
+	/**
+	 * Moves nodes from provided range to target position.
+	 *
+	 * Throws {@link module:utils/ckeditorerror~CKEditorError CKEditorError} `view-writer-invalid-range-container` when
+	 * {@link module:engine/view/range~Range#start start} and {@link module:engine/view/range~Range#end end} positions are not placed inside
+	 * same parent container.
+	 *
+	 * @param {module:engine/view/range~Range} sourceRange Range containing nodes to move.
+	 * @param {module:engine/view/position~Position} targetPosition Position to insert.
+	 * @returns {module:engine/view/range~Range} Range in target container. Inserted nodes are placed between
+	 * {@link module:engine/view/range~Range#start start} and {@link module:engine/view/range~Range#end end} positions.
+	 */
+	move( sourceRange, targetPosition ) {
+		let nodes;
 
-	if ( targetPosition.isAfter( sourceRange.end ) ) {
-		targetPosition = _breakAttributes( targetPosition, true );
+		if ( targetPosition.isAfter( sourceRange.end ) ) {
+			targetPosition = _breakAttributes( targetPosition, true );
 
-		const parent = targetPosition.parent;
-		const countBefore = parent.childCount;
+			const parent = targetPosition.parent;
+			const countBefore = parent.childCount;
 
-		sourceRange = _breakAttributesRange( sourceRange, true );
+			sourceRange = _breakAttributesRange( sourceRange, true );
 
-		nodes = remove( sourceRange );
+			nodes = this.remove( sourceRange );
+
+			targetPosition.offset += ( parent.childCount - countBefore );
+		} else {
+			nodes = this.remove( sourceRange );
+		}
 
-		targetPosition.offset += ( parent.childCount - countBefore );
-	} else {
-		nodes = remove( sourceRange );
+		return this.insert( targetPosition, nodes );
 	}
 
-	return insert( targetPosition, nodes );
-}
+	/**
+     * Wraps elements within range with provided {@link module:engine/view/attributeelement~AttributeElement AttributeElement}.
+     * If a collapsed range is provided, it will be wrapped only if it is equal to view selection.
+     *
+     * If `viewSelection` was set and a collapsed range was passed, if the range is same as selection, the selection
```

Comment:
This comment seems outdated cause `viewSelection` is no longer set in a parameter.